pkg/api/components.go

```diff
@@ -220,7 +220,9 @@ func getComponentDefaultContainerImage(component string, cs *ContainerService) s
 	hyperkubeImageBase := kubernetesImageBase
 	hyperkubeImage := hyperkubeImageBase + k8sComponents[common.Hyperkube]
 	if cs.Properties.IsAzureStackCloud() {
-		hyperkubeImage = hyperkubeImage + common.AzureStackSuffix
+		hyperkubeImage = cs.Properties.OrchestratorProfile.KubernetesConfig.KubernetesImageBase +
```

Comment:
Why do we want to use `cs.Properties.OrchestratorProfile.KubernetesConfig.KubernetesImageBase` instead of `cs.GetCloudSpecConfig().KubernetesSpecConfig.KubernetesImageBase` for Azure Stack?

Reply:
I guess that's part of the question. 

Right now, `cs.GetCloudSpecConfig().KubernetesSpecConfig.KubernetesImageBase` points to gcr but azure stack components should use the custom hyperkube image hosted in MCR.

Would it be better to update `cs.GetCloudSpecConfig().KubernetesSpecConfig.KubernetesImageBase`? I could not easy verify if changing that will break something else.

Reply:
Does `cs.GetCloudSpecConfig().KubernetesSpecConfig.KubernetesImageBase` not point to the `CustomCloudProfile` equivalent? That's what we want, right?

Reply:
It does not, and it is probably not what we need if I understand correctly.
`KubernetesImageBase` is also used as the prefix for addons.

Reply:
I think this is bugz. I think we want `GetCloudSpecConfig()` to return the `AzureEnvironmentSpecConfig` that is inside config object `CustomCloudProfile`. I assume Azure Stack delivers a valid `KubernetesImageBase` value therein, correct?